src/operator.test.js

```diff
@@ -0,0 +1,109 @@
+const { ResourceEventType } = require("@dot-i/k8s-operator");
+const sinon = require("sinon");
+const { expect } = require("./test-setup");
+const Operator = require("./operator");
+const SecretHandler = require("./secret-handler");
+
+describe("operator", () => {
+  beforeEach(() => {
+    this.secretHandlerBackupMock = sinon
+      .stub(SecretHandler.prototype, "backup")
+      .resolves();
+
+    this.operator = new Operator();
+
+    this.watchResourceMock = sinon.stub(this.operator, "watchResource");
+  });
+  afterEach(() => {
+    this.operator.stop();
+    sinon.restore();
+  });
+
+  it("should not call secretHandlerBackupMock if the secret has 'cert-manager.io/certificate-name' annotation", async () => {
+    const invalidTestEvent = {
+      object: {
+        metadata: {
+          annotations: {
+            "cert-manager.io/certificate-name": "tls"
+          }
+        }
+      }
+    };
```

Comment:
This event should have type modified. Without that it does not require code change, as the operator does not backup anything that is not modified.